src/crawlers/cheerio_crawler.js

```diff
@@ -438,6 +440,42 @@ class CheerioCrawler {
         await this.isRunningPromise;
     }
 
+    /**
+     * Function for attaching CrawlerExtensions such as the Unblockers.
+     * @param crawlerExtension - Crawler extension that overrides the crawler configuration.
+     */
+    use(crawlerExtension) {
+        const inheritsFromCrawlerExtension = crawlerExtension instanceof CrawlerExtension;
+
+        if (!inheritsFromCrawlerExtension) {
+            throw new Error('Object passed to the "use" method does not inherit from the "CrawlerExtension" abstract class.');
+        }
+
+        const crawlerExtensionOptions = crawlerExtension.getCrawlerOptions();
+
+        for (const [key, value] of Object.entries(crawlerExtensionOptions)) {
+            const isConfigurable = _.has(this, key);
+            const originalType = typeof this[key];
+            const extensionType = typeof value; // What if we want to null something? It is really needed?
+            const isSameType = originalType === extensionType;
+            const isNill = _.isUndefined(this[key]) || _.isNull(this[key]);
+
+            if (!isConfigurable) { // Test if the property can be configured on the crawler
+                throw new Error(`${crawlerExtension.name} tries to set property "${key}" that is not configurable on CheerioCrawler instance.`);
+            }
+
+            if (!isSameType && !isNill) { // Assuming that extensions will only add up configuration
+                throw new Error(
+                    `${crawlerExtension.name} tries to set property of different type "${extensionType}". "CheerioCrawler.${key}: ${originalType}".`,
+                );
+            }
+
+            defaultLog.warning(`${crawlerExtension.name} is overriding "CheerioCrawler.${key}: ${originalType}" with ${value}.`);
+
+            this[key] = value; // @TODO: Auto binding for functions?
```

Comment:
Depends what you need to bind. Since the function will live on the Crawler now, `this` will be the `crawler` instance.

Reply:
I just realized that you can always bind them in the `getCrawlerOptions` if you need to.